ballerina/tests/12-custom-type-varray-test.bal

```diff
@@ -148,19 +138,268 @@ isolated function insertVarrayWithEmptyArray() returns sql:Error? {
+    check oracledbClient.close();
+}
+
+type InvalidBoolTypeArray record {
+    int pk;
+    boolean[] col_chararr;
+};
+
+@test:Config {
+    groups:["query","varray"],
+    enable: false,
+    dependsOn: [selectVarrayWithInvalidDecimalType]
+}
+isolated function selectVarrayWithInvalidBoolType() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<InvalidBoolTypeArray, sql:Error> streamData = oracledbClient->query(
+        "SELECT pk, COL_CHARARR FROM TestVarrayTable WHERE pk = 1");
+    record {}|error? returnData =  streamData.next();
+    if (returnData is sql:ApplicationError) {
+        test:assertTrue(returnData.message().includes("Cannot cast varray to type: boolean[]"),
+            "Incorrect error message");
+    } else {
+        test:assertFail("Querying varray with invalid array type should fail with " +
+                            "sql:ApplicationError");
+    }
+    check streamData.close();
+    check oracledbClient.close();
+}
+
+type InvalidByteTypeArray record {
+    int pk;
+    byte[] col_chararr;
+};
+
+@test:Config {
+    groups:["query","varray"],
+    enable: false,
+    dependsOn: [selectVarrayWithInvalidDecimalType]
+}
+isolated function selectVarrayWithInvalidByteType() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<InvalidByteTypeArray, sql:Error> streamData = oracledbClient->query(
+        "SELECT pk, COL_CHARARR FROM TestVarrayTable WHERE pk = 1");
+    record {}|error? returnData =  streamData.next();
+    if (returnData is sql:ApplicationError) {
+        test:assertTrue(returnData.message().includes("Cannot cast varray to type: byte[]"),
+            "Incorrect error message");
+    } else {
+        test:assertFail("Querying varray with invalid array type should fail with " +
+                            "sql:ApplicationError");
+    }
+    check streamData.close();
+    check oracledbClient.close();
+}
+
+type InvalidStringTypeArray record {
+    int pk;
+    string[] col_bytearr;
+};
+
+@test:Config {
+    groups:["query","varray"],
+    enable: false,
+    dependsOn: [selectVarrayWithInvalidByteType]
+}
+isolated function selectVarrayWithInvalidStringType() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<InvalidStringTypeArray, sql:Error> streamData = oracledbClient->query(
+        "SELECT pk, COL_BYTEARR FROM TestVarrayTable WHERE pk = 1");
+    record {}|error? returnData =  streamData.next();
+    if (returnData is sql:ApplicationError) {
+        test:assertTrue(returnData.message().includes("Cannot cast varray to type: string[]"),
+            "Incorrect error message");
+    } else {
+        test:assertFail("Querying varray with invalid array type should fail with " +
+                            "sql:ApplicationError");
+    }
+    check streamData.close();
+    check oracledbClient.close();
+}
```

Comment:
```suggestion
}

```

Reply:
fixed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/903f0cd0b25dea933b9106c0947ee5a969e13afd